Runtime/Core/ApiGatewayManagement/AmazonGameLiftClientWrapper.cs

```diff
@@ -1,35 +1,49 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Sockets;
 using System.Threading;
 using System.Threading.Tasks;
 using Amazon.GameLift;
 using Amazon.GameLift.Model;
+using AmazonGameLiftPlugin.Core.CredentialManagement;
+using AmazonGameLiftPlugin.Core.CredentialManagement.Models;
+using AmazonGameLiftPlugin.Core.Shared.FileSystem;
+using UnityEngine;
 
 namespace AmazonGameLiftPlugin.Core.ApiGatewayManagement
 {
     public class AmazonGameLiftClientWrapper : IAmazonGameLiftClientWrapper
     {
         private readonly IAmazonGameLift _amazonGameLiftClient;
-        private static readonly string s_dummyAwsAccessKey = "1";
-        private static readonly string s_dummyAwsSecretKey = "1";
+        private readonly ICredentialsStore _credentialsStore = new CredentialsStore(new FileWrapper());
 
+        private const string FleetDescription = "Created By Amazon GameLift Unity Plugin";
+        
         internal AmazonGameLiftClientWrapper(IAmazonGameLift amazonGameLiftClient)
         {
             _amazonGameLiftClient = amazonGameLiftClient;
         }
 
-        public AmazonGameLiftClientWrapper(string localEndpoint)
+        public AmazonGameLiftClientWrapper(RetriveAwsCredentialsResponse credentialsResponse)
```

Comment:
I'm always wary of passing the Output object as an input to another method directly. Since we already have a interface to create a wrapper using a amazonGameLiftClient can we use that instead?

```
AmazonGameLiftClientWrapper(IAmazonGameLift amazonGameLiftClient)
```

So whoever is creating the wrapper grabs the credentials, creates the client, and then initializes the wrapper with the created client?